src/Reflection/Adapter/ReflectionEnum.php

```diff
@@ -527,17 +530,26 @@ public function getCase(string $name): ReflectionEnumUnitCase|ReflectionEnumBack
         return new ReflectionEnumUnitCase($case);
     }
 
-    /** @return list<ReflectionEnumUnitCase|ReflectionEnumBackedCase> */
+    /** @return list<ReflectionEnumUnitCase>|list<ReflectionEnumBackedCase> */
     public function getCases(): array
     {
-        /** @psalm-suppress ImpureFunctionCall */
-        return array_map(function (BetterReflectionEnumCase $case): ReflectionEnumUnitCase|ReflectionEnumBackedCase {
-            if ($this->betterReflectionEnum->isBacked()) {
-                return new ReflectionEnumBackedCase($case);
+        if ($this->cases !== null) {
+            return $this->cases;
+        }
+
+        $isBacked = $this->betterReflectionEnum->isBacked();
+        $cases    = $this->betterReflectionEnum->getCases();
+
+        $mappedCases = [];
+        foreach ($cases as $case) {
+            if ($isBacked) {
+                $mappedCases[] = new ReflectionEnumBackedCase($case);
+            } else {
+                $mappedCases[] = new ReflectionEnumUnitCase($case);
             }
+        }
 
-            return new ReflectionEnumUnitCase($case);
-        }, array_values($this->betterReflectionEnum->getCases()));
+        return $this->cases = $mappedCases;
```

Comment:
Would likely need to move this to a mutable reference of some sort :thinking: 

I'll think about an approach :-)

Reply:
Yeah, for this particular case, I'll probably write a little `Memoize<T>` utility that is internal to BR

Reply:
this util needs to be a static call, as we can't store an instance of it in the `ReflectionEnum` class, right?

Reply:
ahh got it. give me a minute